tower-buffer/src/service.rs

```diff
@@ -87,13 +87,21 @@ where
     }
 
     fn call(&mut self, request: Request) -> Self::Future {
+        let span = tracing::Span::current();
+        let _guard = span.enter();
```

Comment:
TIOLI: Is it expected that events will occur in the `oneshot::channel` constructor? If not, we can probably just use the span as the parent for the `trace!` call on line 99, and then include it in the message without cloning it.